addons/web_editor/static/src/components/media_dialog/video_selector.js

```diff
@@ -0,0 +1,210 @@
+        };
+
+        this.OPTIONS = {
+            autoplay: {
+                label: this.env._t("Autoplay"),
+                description: this.env._t("Videos are muted when autoplay is enabled"),
+                platforms: [this.PLATFORMS.youtube, this.PLATFORMS.dailymotion, this.PLATFORMS.vimeo],
+                urlParameter: 'autoplay=1',
+            },
+            loop: {
+                label: this.env._t("Loop"),
+                platforms: [this.PLATFORMS.youtube, this.PLATFORMS.vimeo],
+                urlParameter: 'loop=1',
+            },
+            hide_controls: {
+                label: this.env._t("Hide player controls"),
+                platforms: [this.PLATFORMS.youtube, this.PLATFORMS.dailymotion],
+                urlParameter: 'controls=0',
+            },
+            hide_fullscreen: {
+                label: this.env._t("Hide fullscreen button"),
+                platforms: [this.PLATFORMS.youtube],
+                urlParameter: 'fs=0',
+                isHidden: () => this.state.options.filter(option => option.id === 'hide_controls')[0].value,
+            },
+            hide_yt_logo: {
+                label: this.env._t("Hide Youtube logo"),
+                platforms: [this.PLATFORMS.youtube],
+                urlParameter: 'modestbranding=1',
+                isHidden: () => this.state.options.filter(option => option.id === 'hide_controls')[0].value,
+            },
+            hide_dm_logo: {
+                label: this.env._t("Hide Dailymotion logo"),
+                platforms: [this.PLATFORMS.dailymotion],
+                urlParameter: 'ui-logo=0',
+            },
+            hide_dm_share: {
+                label: this.env._t("Hide sharing button"),
+                platforms: [this.PLATFORMS.dailymotion],
+                urlParameter: 'sharing-enable=0',
+            },
+        };
+
+        this.state = useState({
+            options: [],
+            src: '',
+            urlInput: '',
+            platform: null,
+            vimeoPreviews: [],
+        });
+        this.urlInputRef = useRef('url-input');
+
+        useEffect(() => {
+            const initWithMedia = async () => {
+                if (this.props.media) {
+                    const src = this.props.media.dataset.oeExpression || this.props.media.dataset.src || (this.props.media.tagName === 'IFRAME' && this.props.media.getAttribute('src')) || '';
+                    if (src) {
+                        this.urlInputRef.el.value = src;
+                        await this.updateVideo(src);
+
+                        this.state.options = this.state.options.map((option) => {
+                            const { urlParameter } = this.OPTIONS[option.id];
+                            return { ...option, value: src.indexOf(urlParameter) >= 0 };
+                        });
+                    }
+                }
+            };
+
+            initWithMedia();
+        }, () => []);
+
+        useEffect(() => {
```

Comment:
Same debate as above, probably use only `onMounted`? Especially for multiple `useEffect` with the same (empty) dependency? :grimacing: